From 6c2f967d3093ed1fa4335273ec44d15a2e53c9be Mon Sep 17 00:00:00 2001
From: Liu Changhui <changhui.liu@windriver.com>
Date: Fri, 29 Jan 2010 13:27:46 +0800
Subject: [PATCH] Platform: RMI arch Kconfig and Makefile support

RMI arch Kconfig and Makefile support.Add some macro defination for
RMI kernel build.

source: from RMI SDK1.7

Signed-off-by: shuo.kang <shuo.kang@windriver.com>
---
 arch/mips/Kconfig  |   38 ++++++++++++++++++++++++++++++++++++++
 arch/mips/Makefile |   22 ++++++++++++++++++++++
 2 files changed, 60 insertions(+), 0 deletions(-)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 37ef41e..2c75827 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -598,6 +598,30 @@ config WR_PPMC
 	  This enables support for the Wind River MIPS32 4KC PPMC evaluation
 	  board, which is based on GT64120 bridge chip.
 
+config RMI_PTR
+	bool "Support for RMI ATX board"
+	depends on EXPERIMENTAL
+	select SMP
+	select BOOT_ELF32
+	select RMI_PHOENIX
+	select SYS_HAS_CPU_PHOENIX
+	select SYS_SUPPORTS_SMP
+	select HW_HAS_PCI
+	select SWAP_IO_SPACE
+	select SYS_SUPPORTS_32BIT_KERNEL
+	select SYS_SUPPORTS_64BIT_KERNEL
+	select SYS_SUPPORTS_BIG_ENDIAN
+	select SYS_SUPPORTS_HIGHMEM
+	select DMA_COHERENT
+	select CEVT_R4K
+	select CSRC_R4K
+	select IRQ_CPU
+	select ZONE_DMA if 64BIT
+	select SYNC_R4K
+	help
+	  This board is based on RMI XLS processor. 
+	  Say Y here to support this machine type
+
 endchoice
 
 source "arch/mips/au1000/Kconfig"
@@ -609,6 +633,7 @@ source "arch/mips/sgi-ip27/Kconfig"
 source "arch/mips/sibyte/Kconfig"
 source "arch/mips/txx9/Kconfig"
 source "arch/mips/vr41xx/Kconfig"
+source "arch/mips/rmi/Kconfig"
 
 endmenu
 
@@ -1176,6 +1201,16 @@ config CPU_SB1
 	select CPU_SUPPORTS_HIGHMEM
 	select WEAK_ORDERING
 
+config CPU_PHOENIX
+	bool "XLS"
+	depends on SYS_HAS_CPU_PHOENIX
+	select CPU_SUPPORTS_32BIT_KERNEL
+	select CPU_SUPPORTS_64BIT_KERNEL
+	select CPU_SUPPORTS_HIGHMEM
+	select CPU_HAS_LLSC
+	select WEAK_ORDERING
+	select WEAK_REORDERING_BEYOND_LLSC
+
 endchoice
 
 config SYS_HAS_CPU_LOONGSON2
@@ -1247,6 +1282,9 @@ config SYS_R4K_CVET_HWTIMER
 config WEAK_ORDERING
 	bool
 
+config SYS_HAS_CPU_PHOENIX
+	bool
+
 #
 # CPU may reorder reads and writes beyond LL/SC
 # CPU may reorder R->LL, R->LL, W->LL, W->LL, R->SC, R->SC, W->SC, W->SC
diff --git a/arch/mips/Makefile b/arch/mips/Makefile
index 84b5241..0c0bd5f 100644
--- a/arch/mips/Makefile
+++ b/arch/mips/Makefile
@@ -145,6 +145,7 @@ cflags-$(CONFIG_CPU_R8000)	+= -march=r8000 -Wa,--trap
 cflags-$(CONFIG_CPU_R10000)	+= $(call cc-option,-march=r10000,-march=r8000) \
 			-Wa,--trap
 
+cflags-$(CONFIG_CPU_PHOENIX)    += $(call cc-option,-march=r8000) -Wa,--trap -DXLS -DRMI_BRIDGE_WKAROUND
 cflags-$(CONFIG_CPU_R4000_WORKAROUNDS)	+= $(call cc-option,-mfix-r4000,)
 cflags-$(CONFIG_CPU_R4400_WORKAROUNDS)	+= $(call cc-option,-mfix-r4400,)
 cflags-$(CONFIG_CPU_DADDI_WORKAROUNDS)	+= $(call cc-option,-mno-daddi,)
@@ -579,7 +580,24 @@ core-$(CONFIG_TOSHIBA_RBTX4927)	+= arch/mips/txx9/rbtx4927/
 #
 core-$(CONFIG_TOSHIBA_RBTX4938) += arch/mips/txx9/rbtx4938/
 
+#
+# RMI Phoenix SOC
+core-$(CONFIG_RMI_PHOENIX)      += arch/mips/rmi/phoenix/ 
+
+#
+# RMI PTR board
+core-$(CONFIG_RMI_PTR)          += arch/mips/rmi/ptr/
+cflags-$(CONFIG_RMI_PTR)	+= -Iinclude/asm-mips/mach-rmi
+cflags-$(CONFIG_CRYPTO_XLR)	+= -Idrivers/crypto/rmi/common
+
 cflags-y			+= -Iinclude/asm-mips/mach-generic
+
+
+#
+# This address is now configured via kernel configuration file
+#
+load-$(CONFIG_RMI_PTR)          += $(CONFIG_RMI_PHOENIX_LOAD_ADDRESS)
+
 drivers-$(CONFIG_PCI)		+= arch/mips/pci/
 
 ifdef CONFIG_PHYSICAL_START
@@ -703,6 +721,10 @@ ifdef CONFIG_MIPS32_O32
 	@echo '  Checking missing-syscalls for O32'
 	$(Q)$(MAKE) $(build)=. missing-syscalls EXTRA_CFLAGS="-mabi=32"
 endif
+ifdef CONFIG_RMI_PHOENIX
+	@echo ' Building message ring configuration'
+	$(MAKE) -C  $(srctree)/arch/mips/rmi/phoenix msgring
+endif
 
 archclean:
 	@$(MAKE) $(clean)=arch/mips/boot
-- 
1.6.0.4

